crates/hir-ty/src/consteval.rs

```diff
@@ -269,13 +277,17 @@ pub(crate) fn const_eval_discriminant_variant(
         };
         return Ok(value);
     }
+
+    let repr = db.enum_data(loc.parent).repr;
+    let is_signed = repr.and_then(|repr| repr.int).is_none_or(|int| int.is_signed());
```

Comment:
I think the default `repr` might be `isize`?

Reply:
Yes (https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=eadf4c6522ff59e875059b28b4cae221).